cmake-modules/AzureDoxygen.cmake

```diff
@@ -19,7 +19,15 @@ function(generate_documentation PROJECT_NAME PROJECT_VERSION)
         set(DOXYGEN_RECURSIVE YES)
         set(DOXYGEN_USE_MDFILE_AS_MAINPAGE ./README.md)
         set(DOXYGEN_INLINE_SOURCES YES)
+        # Skip generating docs for json, test and private files.
+        set(DOXYGEN_EXCLUDE_PATTERNS
+            json.hpp
+            test
+            *private*)
+        # Skip documenting Details symbols (all from ::Details namespace)
+        set(DOXYGEN_EXCLUDE_SYMBOLS Details)
```

Comment:
We should probably exclude `Internal` too.

Reply:
Internal is something developers will be interested in.  I would probably, at least, add an CMake option to show internal docs with default value OFF.
Then some service developer could turn in ON to see docs

Reply:
doxygen docs and api docs are **public** facing and for end customers. Since we are by source, any client library author can already see the doc strings for internal types and methods if they need to. I don't think it is that useful to have a cmake option to turn those on/off because I doubt SDK developers and contributors read the generated docs for their decision making.